### PR TITLE
Add property-based fuzz test

### PR DESCRIPTION
**Pull request by MaxG87**
A test is added that runs mccabe against plenty of random yet valid
Python source code. Right now, the test does not do much - it only tests
whether mccabe accepts the code. Already this is an achievement because
now weird edge cases are covered.

Having this test suite integrated will also allow to build up on that.
The author of Hypothesis claimed to work on a coverage guided test
runner. Also more general properties to hold on any valid source code
now can be tested.

### CODE REVIEW

File: test_mccabe.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import sys
+
 try:
     from StringIO import StringIO
 except ImportError:
     from io import StringIO
 
 import pytest
+import hypothesmith
+from hypothesis import HealthCheck, given, settings, strategies as st
```
Comment (cclauss) [inline test_mccabe.py:11]:
```suggestion
try:
    import hypothesmith
    from hypothesis import HealthCheck, given, settings, strategies as st
except ImportError:
    hypothesmith = None
```

File: test_mccabe.py [hunk 2]
```diff
@@ -233,5 +236,40 @@ def test_get_module_complexity(self):
...
+    suppress_health_check=HealthCheck.all(),  # this is slow and filter-heavy.
+)
+@given(
+    # Note that while Hypothesmith might generate code unlike that written by
+    # humans, it's a general test that should pass for any *valid* source code.
+    # (so e.g. running it against code scraped of the internet might also help)
+    src_contents=hypothesmith.from_grammar() | hypothesmith.from_node(),
+    max_complexity=st.integers(min_value=1),
+)
+def test_idempotent_any_syntatically_valid_python(
```
Comment (cclauss) [inline test_mccabe.py:254]:
```suggestion
@pytest.mark.skipif(not hypothesmith, reason="hypothesmith could not be imported")
                                  def test_idempotent_any_syntatically_valid_python(
```
https://docs.pytest.org/en/reorganize-docs/new-docs/user/skipping.html

**Comment by MaxG87** [comment]
Great! Thank you very much. If you would send me a patch I would apply it so you get credits and authorship.

**Comment by cclauss** [comment]
I do not know how to generate a patch and I do not need credit so please try to apply these changes on your end.  My sense is that `given` and `settings` might still give you trouble.  If so, you can stub them out or `@skipif` a whole class that contains your test code.

**Comment by sigmavirus24** [comment]
@MaxG87 if you use the GitHub UI you "commit(ting) suggestion" or "add(ing) suggestion to batch" will give @cclauss credit

**Review by sigmavirus24 — APPROVED** [review]

**Review by cclauss — APPROVED** [review]

**Comment by sigmavirus24** [comment]
Closing and re-opening to trigger a new merged build on travis
